frontend/src/components/Editor/Editor.js

```diff
@@ -268,64 +249,40 @@ const Editor = ({
   };
 
-
-  const handleFormattingError = (fileName) => {
-    const errorMessage =
-      fileName === "indexingLogic.js"
-        ? "Oh snap! We could not format your code. Make sure it is proper Javascript code."
-        : "Oh snap! We could not format your SQL schema. Make sure it is proper SQL DDL";
-
-    setError(() => errorMessage);
-  };
-
   const reformatAll = (indexingCode, schema) => {
-    const formattedCode = formatIndexingCode(indexingCode);
+    let formattedCode = indexingCode
+    let formattedSql = schema;
+    try {
+      formattedCode = formatIndexingCode(indexingCode);
+      setCodeFormattingError(() => undefined);
+    } catch (error) {
+      console.log("error", error)
+      setCodeFormattingError(() => "Oh snap! We could not format your code. Make sure it is proper Javascript code.");
+    }
+    try {
+      formattedSql = formatSQL(schema);
+      setSchemaFormattingError(() => undefined);
+    } catch (error) {
+      setSchemaFormattingError(() => "Oh snap! We could not format your SQL schema. Make sure it is proper SQL DDL");
+    }
     setIndexingCode(formattedCode);
-
-    const formattedSchema = formatSQL(schema);
-    setSchema(formattedSchema);
-
-    return { formattedCode, formattedSchema }
-  }
-
-  const reformat = (indexingCode, schema) => {
-    return new Promise((resolve, reject) => {
-      try {
-        let formattedCode;
-        if (fileName === "indexingLogic.js") {
-          formattedCode = formatIndexingCode(indexingCode);
-          setIndexingCode(formattedCode);
-        } else if (fileName === "schema.sql") {
-          formattedCode = formatSQL(schema);
-          setSchema(formattedCode);
-        }
-        setError(() => undefined);
-        resolve(formattedCode);
-      } catch (error) {
-        handleFormattingError(fileName);
-        reject(error);
-      }
-    });
+    setSchema(formattedSql);
+    return { formattedCode, formattedSql }
   };
 
   function handleCodeGen() {
     try {
       setSchemaTypes(pgSchemaTypeGen.generateTypes(schema));
       attachTypesToMonaco(); // Just in case schema types have been updated but weren't added to monaco
-      setError(() => undefined);
-    } catch (error) {
-      handleCodeGenError(error);
+      setTypesGenerationError(() => undefined);
+    } catch (typesError) {
+      console.log("typesGenerationError generating types for saved schema.\n", typesError);
+      setTypesGenerationError(() => "Oh snap! We could not generate types for your SQL schema. Make sure it is proper SQL DDL.");
     }
   }
 
-  const handleCodeGenError = (error) => {
-    console.error("Error generating types for saved schema.\n", error);
-    const errorMessage = "Oh snap! We could not generate types for your SQL schema. Make sure it is proper SQL DDL."
-    setError(() => errorMessage);
-  };
-
   async function handleFormating() {
-    await reformat(indexingCode, schema);
+    await reformatAll(indexingCode, schema);
```

Comment:
```suggestion
    reformatAll(indexingCode, schema);
```
This function is not `async`